sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -393,4 +404,15 @@ case class DataSourceResolution(
       nullable = true,
       builder.build())
   }
+
+  private def requireNamespaceExists(
+      catalog: CatalogPlugin,
+      namespace: Option[Seq[String]]): Unit = {
+    namespace.map { ns =>
+      val nsArray = ns.toArray
+      if (!catalog.asNamespaceCatalog.namespaceExists(nsArray)) {
```

Comment:
This means that we can't set current namespace for a v2 catalog that doesn't implement `SupportsNamespace`.

This is a little counter-intuitive as `TableCatalog.loadTable` does accept a namespace parameter. Maybe we shouldn't do the namespace check when setting current namespace.

Reply:
I removed this check. For V1, database check will be performed in `SessionCatalog.setCurrentDatabase` (so the behavior remains the same) and for V2, there will be no checks performed.